### PR TITLE
Add docs for internal Nextstrain team

### PR DESCRIPTION
**Pull request by joverlee521**
## Description of proposed changes

I didn't want to take the time to set up of full Read the Docs project since I'm not even sure that's needed. 
Just using Markdown files to document:

1. the CI workflow
2. automating pathogen repo workflows

## Related issue(s)

Resolves https://github.com/nextstrain/pathogen-repo-guide/issues/10

## Checklist

- [ ] Checks pass

### CODE REVIEW

File: docs/nextstrain-internal/pathogen-repo-ci.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+### `pathogen-repo-ci` Requirements
+
+> [!IMPORTANT]
+> If your pathogen repo does not meet these requirements, please update the
+> repo to conform to them to maintain uniformity of Nextstrain pathogen repos.
+
+1. The pathogen repo must have a top level `nextstrain-pathogen.yaml` file.
+2. To run the CI step for a workflow (ingest, nextclade, or phylogenetic), the
+directory must contain a `build-configs/ci/config.yaml` config file.
+3. At least one of the CI steps must be run or else the CI workflow will exit with an error.
```
Comment (joverlee521) [inline docs/nextstrain-internal/pathogen-repo-ci.md:38]: I waffled on whether this should be here or point to docs in the `.github` repo. I just put it here so it's all in one place, but feel free to push back.

Comment (jameshadfield) [inline docs/nextstrain-internal/pathogen-repo-ci.md:38]: What are "steps" in this context? Presumably not strictly the action steps of the 'nextstrain-build' as (e.g.) the `Log in to docker.io` step will always run, yeah?

File: docs/nextstrain-internal/pathogen-repo-ci.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+```yaml
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
```
Comment (tsibley) [inline docs/nextstrain-internal/pathogen-repo-ci.md:22]: I'd encourage adding a basic `schedule:` on here too. It's really nice to come back to a repo after a while of inactivity and know if CI has broken due to the passage of time before you do some work and expect CI to pass.

_non-blocking_

File: docs/nextstrain-internal/pathogen-repo-ci.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+jobs:
+  ci:
+    uses: nextstrain/.github/.github/workflows/pathogen-repo-ci.yaml@master
```
Comment (tsibley) [inline docs/nextstrain-internal/pathogen-repo-ci.md:26]: Presumably once we tag v1 (when? I'd have thought by now?) we'll suggest that here?

_non-blocking_

**Review by tsibley — APPROVED** [review]

File: docs/nextstrain-internal/pathogen-repo-ci.md (hunk 3 above)
Comment (genehack) [reply docs/nextstrain-internal/pathogen-repo-ci.md:26]: Tagging the repo is on my TODO list; I was waiting for the "smart" `pathogen-repo-ci` to land, which just happened yesterday (and it looks like you've got at least one further outstanding change).

**Review by genehack — APPROVED** [review]

File: docs/nextstrain-internal/pathogen-repo-ci.md (hunk 2 above)
Comment (joverlee521) [reply docs/nextstrain-internal/pathogen-repo-ci.md:22]: > I'd encourage adding a basic schedule: on here too. It's really nice to come back to a repo after a while of inactivity and know if CI has broken due to the passage of time before you do some work and expect CI to pass.

True! What would be a reasonable schedule here? Once a week?

Comment (joverlee521) [reply docs/nextstrain-internal/pathogen-repo-ci.md:22]: Added in https://github.com/nextstrain/pathogen-repo-guide/pull/45/commits/b183592127529e036785db5aff1a5a2774b57414

**Pull request merged**